openrr-teleop/src/control_nodes_config.rs

```diff
@@ -115,6 +150,228 @@ impl ControlNodesConfig {
             }
         }
 
-        nodes
+        Ok(nodes)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use arci::{
+        DummyJointTrajectoryClient, DummyLocalization, DummyMoveBase, DummyNavigation, DummySpeaker,
+    };
+    use openrr_client::{
+        create_random_jacobian_ik_solver, IkSolverParameters, IkSolverWithChain,
+        OpenrrClientsConfig,
+    };
+
+    use super::*;
+    use crate::{default_move_step_angular, default_move_step_linear, default_step_duration_secs};
+
+    #[test]
+    fn test_gen() {
+        // JoyJointTeleopConfig
+        let config = JoyJointTeleopNodeConfig {
+            mode: String::from("a"),
+            joint_step: 1.0_f64,
+            step_duration_secs: default_step_duration_secs(),
+        };
+        let _joy_tconfig = JoyJointTeleopConfig {
+            client_name: String::from("b"),
+            config,
+        };
+
+        // IkNodeTeleopConfig
+        let config = IkNodeConfig {
+            mode: String::from("c"),
+            move_step_angular: default_move_step_angular(),
+            move_step_linear: default_move_step_linear(),
+            step_duration_secs: default_step_duration_secs(),
+        };
+
+        let _iknode_tconfig = IkNodeTeleopConfig {
+            solver_name: String::from("d"),
+            joint_trajectory_client_name: String::from("f"),
+            config,
+        };
+    }
+
+    #[test]
+    fn test_control_node_config_create_error() {
+        let jj_config = JoyJointTeleopConfig {
+            client_name: String::from("jj config"),
+            config: JoyJointTeleopNodeConfig {
+                mode: String::from("jjt config"),
+                joint_step: 1.0_f64,
+                step_duration_secs: default_step_duration_secs(),
+            },
+        };
+        let ik_config = IkNodeTeleopConfig {
+            solver_name: String::from("ik config solver"),
+            joint_trajectory_client_name: String::from("ik trajectory"),
+            config: IkNodeConfig {
+                mode: String::from("ik node config"),
+                move_step_angular: default_move_step_angular(),
+                move_step_linear: default_move_step_linear(),
+                step_duration_secs: default_step_duration_secs(),
+            },
+        };
+        let rc_config = RobotCommandConfig {
+            name: String::from("rc config"),
+            file_path: String::from("sample path"),
+        };
+        let ctrl_node_config = ControlNodesConfig {
+            move_base_mode: Some(String::from("a")),
+            joy_joint_teleop_configs: vec![jj_config],
+            ik_node_teleop_configs: vec![ik_config],
+            joints_pose_sender_config: None,
+            command_configs: vec![rc_config],
+        };
+
+        // Check create control node(Error:HashMap length)
```

Comment:
Is this right?
It seems to be a matter of whether the value for the specified key exists in the HashMap or not, rather than the length.

Reply:
I'm sorry that I think length 0 same as not contained key.
But `specified key exists in the HashMap or not` is more correct.